sys/include/net/af.h

```diff
@@ -47,5 +48,9 @@ enum {
 }
 #endif
 
+#else /* BOARD_NATIVE */
+#include <arpa/inet.h>
```

Comment:
I think it is, as native uses native libraries, which do not necessarily share RIOT's idea of e.g., AF_INET. So even if code compiles, we need to make sure RIOT's net/af.h is never used together with the system libs.